hyperhtml.js

```diff
@@ -193,321 +156,692 @@ var hyperHTML = (function () {'use strict';
+            parseFloat(RegExp.$1) < 55;
+
+  // If attributes order is shuffled, threat the browser differently
+  // Usually this is a well known IE only limitation but some older FF does the same.
+  var IE =  (function (p) {
+              p.innerHTML = '<i data-i="" class=""></i>';
+              return /class/i.test(p.firstChild.attributes[0].name);
+            }(document.createElement('p')));
+
+  // ---------------------------------------------
   // Helpers
-  // -------------------------
+  // ---------------------------------------------
 
-  // given a generic text or comment node
-  // remove the surrounding emptiness
-  function cleanAround(textNode, sibling) {
-    var adjacent = textNode[sibling];
-    if (
-      adjacent &&
-      adjacent.nodeType === TEXT_NODE &&
-      trim.call(adjacent.textContent).length < 1
+  // used to convert childNodes to Array
+  var slice = [].slice;
+
+  // return a single node or an Array or nodes
+  function createContent(node) {
+    for (var
+      child,
+      content = [],
+      childNodes = node.childNodes,
+      i = 0,
+      length = childNodes.length;
+      i < length; i++
     ) {
-      textNode.parentNode.removeChild(adjacent);
-      cleanAround(textNode, sibling);
+      child = childNodes[i];
+      if (
+        child.nodeType === ELEMENT_NODE ||
+        0 < trim.call(child.textContent).length
+      ) {
+        content.push(child);
+      }
     }
+    return content.length === 1 ? content[0] : content;
   }
 
-  // find a live node through a virtual one
-  function find(live, virtual) {
-    var i, length, name, parentNode, map = [];
-    switch(virtual.nodeType) {
-      case ELEMENT_NODE:
-        parentNode = virtual;
-        break;
-      case COMMENT_NODE:
-        // clean up empty text nodes around
-        cleanAround(virtual, 'previousSibling');
-        cleanAround(virtual, 'nextSibling');
-        parentNode = virtual.parentNode;
-        map.unshift('childNodes', map.indexOf.call(parentNode.childNodes, virtual));
-        break;
-      case ATTRIBUTE_NODE:
-      default: // jsdom here does not provide a nodeType 2 ...
-        parentNode = virtual.ownerElement;
-        map.unshift('getAttributeNode', virtual.name);
-        break;
+  // given a node, inject some html and return
+  // the resulting template document fragment
+  function createFragment(node, html) {
+    var fragment;
+    var document = node.ownerDocument;
+    var container = document.createElement('template');
+    var hasContent = 'content' in container;
+    var fallback = false;
+    if (!hasContent) {
+      fragment = createDocumentFragment(document);
+      fallback = /^[^\S]*?<(t(?:head|body|foot|r|d|h))/i.test(html);
     }
-    virtual = parentNode;
-    while (parentNode = parentNode.parentNode) {
-      map.unshift('children', map.indexOf.call(parentNode.children, virtual));
-      virtual = parentNode;
+    if (fallback) {
```

Comment:
OK, I'll put a comment but this is a fallback in two cases: there is no HTMLTemplateElement in the engine, the content is a partial table internal content that cannot be created via innerHTML

tableFallback would be even more deceptive, because it's a "don't know about tempalte" problem too.